<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <!--
        Parameters
    -->
    
    <arg name="quad" default="SQ01s"/>
    <arg name="is_sim" default="true"/>

    <arg name="is_constant_drift" default="true"/>
    <arg name="constant_drift_x" default="1.0"/>
    <arg name="constant_drift_y" default="0.0"/>
    <arg name="constant_drift_z" default="0.0"/>
    <arg name="constant_drift_roll" default="0.0"/>
    <arg name="constant_drift_pitch" default="0.0"/>
    <arg name="constant_drift_yaw" default="0.0"/>
   
    <arg name="is_linear_drift" default="false"/>
    <arg name="linear_drift_rate_x" default="0.0"/>
    <arg name="linear_drift_rate_y" default="0.0"/>
    <arg name="linear_drift_rate_z" default="0.0"/>
    <arg name="linear_drift_rate_roll" default="0.0"/>
    <arg name="linear_drift_rate_pitch" default="0.0"/>
    <arg name="linear_drift_rate_yaw" default="0.0"/>

    <!--
        Nodes
    -->

    <node ns="$(arg quad)" name="pose_corrupter" pkg="primer" type="pose_corrupter.py" output="screen">
        <param name="is_sim" value="$(arg is_sim)"/>
        <param name="is_constant_drift" value="$(arg is_constant_drift)"/>
        <param name="constant_drift_x" value="$(arg constant_drift_x)"/>
        <param name="constant_drift_y" value="$(arg constant_drift_y)"/>
        <param name="constant_drift_z" value="$(arg constant_drift_z)"/>
        <param name="constant_drift_roll" value="$(arg constant_drift_roll)"/>
        <param name="constant_drift_pitch" value="$(arg constant_drift_pitch)"/>
        <param name="constant_drift_yaw" value="$(arg constant_drift_yaw)"/>
        <param name="is_linear_drift" value="$(arg is_linear_drift)"/>
        <param name="linear_drift_rate_x" value="$(arg linear_drift_rate_x)"/>
        <param name="linear_drift_rate_y" value="$(arg linear_drift_rate_y)"/>
        <param name="linear_drift_rate_z" value="$(arg linear_drift_rate_z)"/>
        <param name="linear_drift_rate_roll" value="$(arg linear_drift_rate_roll)"/>
        <param name="linear_drift_rate_pitch" value="$(arg linear_drift_rate_pitch)"/>
        <param name="linear_drift_rate_yaw" value="$(arg linear_drift_rate_yaw)"/>
    </node>

</launch>